### PR TITLE
Fix Iframe.Credentialless documentation.

### PR DESCRIPTION
**Pull request by ArthurSonzogni**
### Description

I would like to apply some minor edit to the original PR introducing `Iframe.credentialless`:
https://github.com/mdn/content/pull/23097/files
@chrisdavidmills kindly offered me reviewing his original document and PR, but I missed it, being on vacation during Christmas.

### Motivation
- Previously, the document explained `This context does not have access to the parent context's shared storage and credentials`. This might be a bit misleading, because in the context of an iframe, the parent often refers to the document embedded in the iframe. It is unrelated here.  It would be better to explain the document don't have access to the network/cookies/storage partition associated with the document's origin.
- In a few location, make it clear the new context is local to the top-level document.
- Add a references to `COEP:credentialless`.

### Additional details
- https://wicg.github.io/anonymous-iframe/

### Related issues and pull requests
- https://github.com/mdn/content/pull/23097/files

### CODE REVIEW

**Comment by ArthurSonzogni** [comment]
@chrisdavidmills and @sideshowbarker. What do you think of this?

File: files/en-us/web/api/window/credentialless/index.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ browser-compat: api.Window.credentialless
 
 The **`window.credentialless`** read-only property returns a boolean that indicates whether the current document was loaded inside a credentialless {{htmlelement("iframe")}}, meaning that it is loaded in a new, ephemeral context.
 
-This context does not have access to the parent context's shared storage and credentials. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not.
+This context doesn't not have access to its network, cookies, and storage data associated with its origin. It uses a new context local to the top-level document lifetime. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not.
```
Comment (chrisdavidmills) [inline files/en-us/web/api/window/credentialless/index.md:21]:
```suggestion
This context doesn't have access to the network, cookies, and storage data associated with its origin. It uses a new context local to the top-level document lifetime. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not.
```

File: files/en-us/web/html/element/iframe/index.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ This element includes the [global attributes](/en-US/docs/Web/HTML/Global_attrib
 
 - {{htmlattrdef("credentialless")}} {{Experimental_Inline}} {{Non-standard_Inline}}
 
-  - : Set to `true` to make the `<iframe>` credentialless, meaning that its content is loaded in a new, ephemeral context. This context does not have access to the parent context's shared storage and credentials. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not. See [IFrame credentialless](/en-US/docs/Web/Security/IFrame_credentialless) for more details.
+  - : Set to `true` to make the `<iframe>` credentialless, meaning that its content will be loaded in a new, ephemeral context. It doesn't not have access to the network, cookies and storage data associated with its origin. The context is a new one, local to the current top-level document lifetime. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not. See [IFrame credentialless](/en-US/docs/Web/Security/IFrame_credentialless) for more details.
```
Comment (chrisdavidmills) [inline files/en-us/web/html/element/iframe/index.md:54]:
```suggestion
  - : Set to `true` to make the `<iframe>` credentialless, meaning that its content will be loaded in a new, ephemeral context. It doesn't have access to the network, cookies, and storage data associated with its origin. It uses a new context local to the top-level document lifetime. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not. See [IFrame credentialless](/en-US/docs/Web/Security/IFrame_credentialless) for more details.
```

**Review by chrisdavidmills — COMMENTED** [review]
A few language nitpicks from me, but mostly looking good. Thanks @ArthurSonzogni !

File: files/en-us/web/api/htmliframeelement/credentialless/index.md [hunk 3]
```diff
@@ -15,9 +15,11 @@ browser-compat: api.HTMLIFrameElement.credentialless
 
 {{APIRef("HTML DOM")}}{{SeeCompatTable}}
 
-The **`credentialless`** property of the {{domxref("HTMLIFrameElement")}} interface indicates whether the {{htmlelement("iframe")}} is credentialless, meaning that its content is loaded in a new, ephemeral context.
+The **`credentialless`** property of the {{domxref("HTMLIFrameElement")}} interface indicates whether the {{htmlelement("iframe")}} is credentialless, meaning that documents inside will be loaded using new, ephemeral contexts.
 
-This context does not have access to the parent context's shared storage and credentials. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not. See [IFrame credentialless](/en-US/docs/Web/Security/IFrame_credentialless) for a deeper explanation.
+Those contexts do not have access to their network, cookies and storage data associated with their origin. Instead, they use new ones, local to the top-level document lifetime. It means any data stored won't be accessible anymore after the user's navigate away from the page or reloads it.
```
Comment (sideshowbarker) [inline files/en-us/web/api/htmliframeelement/credentialless/index.md:20]:
```suggestion
Those contexts do not have access to their network, cookies and storage data associated with their origin. Instead, they use new ones, local to the top-level document lifetime. It means any data stored won't be accessible anymore after the user navigates away from the page or reloads it.
```

File: files/en-us/web/api/htmliframeelement/credentialless/index.md [hunk 4]
```diff
@@ -15,9 +15,11 @@ browser-compat: api.HTMLIFrameElement.credentialless
...
+
+In return the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not. See [IFrame credentialless](/en-US/docs/Web/Security/IFrame_credentialless) for a deeper explanation.
```
Comment (sideshowbarker) [inline files/en-us/web/api/htmliframeelement/credentialless/index.md:22]:
```suggestion
In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not. See [IFrame credentialless](/en-US/docs/Web/Security/IFrame_credentialless) for a deeper explanation.
```

File: files/en-us/web/security/iframe_credentialless/index.md [hunk 5]
```diff
@@ -52,9 +52,9 @@ iframeElem.src = 'https://en.wikipedia.org/wiki/Spectre_(security_vulnerability)
...
+This results in the documents inside the credentialless `<iframe>` being loaded using new, ephemeral contexts — those contexts don't have access to the data associated with their origins, for example [cookies](/en-US/docs/Web/HTTP/Cookies) and [localStorage](/en-US/docs/Web/API/Window/localStorage). The credentialless storage is partitioned out separately with storage keys modified by a nonce ("number used once") value, set once per top-level document. So a cookie set in one credentialless `<iframe>` will be accessible only from other same-origin credentialless `<iframe>`s embedded below the same top-level document.
```
Comment (sideshowbarker) [inline files/en-us/web/security/iframe_credentialless/index.md:55]:
```suggestion
This results in the documents inside the credentialless `<iframe>` being loaded using new, ephemeral contexts — those contexts don't have access to the data associated with their origins; for example [cookies](/en-US/docs/Web/HTTP/Cookies) and [localStorage](/en-US/docs/Web/API/Window/localStorage). The credentialless storage is partitioned out separately with storage keys modified by a nonce ("number used once") value, set once per top-level document. So a cookie set in one credentialless `<iframe>` will be accessible only from other same-origin credentialless `<iframe>`s embedded below the same top-level document.
```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
Arthur, nice work on this update, and congrats on landing your first direct docs change here — welcome aboard 🎉

**Comment by ArthurSonzogni** [comment]
Thank you!

File: files/en-us/web/security/index.md [hunk 6]
```diff
@@ -51,7 +51,7 @@ Ensuring that your website or open web application is secure is critical. Even s
...
   - : There are privacy and security risks associated with the [Referer HTTP header](/en-US/docs/Web/HTTP/Headers/Referer). This article describes them and offers advice on mitigating those risks.
 - [IFrame credentialless](/en-US/docs/Web/Security/IFrame_credentialless)
-  - : Iframe credentialless provides a mechanism for developers to load third-party resources in {{htmlelement("iframe")}}s using a new, ephemeral context that does not have access to the parent context's shared storage and credentials. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not.
+  - : Iframe credentialless provides a mechanism for developers to load third-party resources in {{htmlelement("iframe")}}s using a new, ephemeral context. This context doesn't have access to the network, cookies, and storage data associated with is origin. It uses a new context local to the top-level document lifetime. In return, the {{httpheader("Cross-Origin-Embedder-Policy")}} (COEP) embedding rules can be lifted, so documents with COEP set can embed third-party documents that do not.
```
Comment (noman-land) [inline files/en-us/web/security/index.md:54]: Hello, I came across this documentation and this bit confused me. I believe there may be a typo here:

"and storage data associated with **is** origin"

Perhaps this should be "with **its** origin"?

Comment (sideshowbarker) [reply files/en-us/web/security/index.md:54]: @noman-land 

> "and storage data associated with **is** origin"
> 
> Perhaps this should be "with **its** origin"?

Yes — thanks much for catching that.

Can you create a PR with that fix? If not, that’s fine — I could create a PR for it myself, if you’d prefer.

Comment (noman-land) [reply files/en-us/web/security/index.md:54]: Will do today!

Comment (noman-land) [reply files/en-us/web/security/index.md:54]: Submitted PR here: https://github.com/mdn/content/pull/24371

**Pull request merged**